Homogeneous_kernel/include/CGAL/Homogeneous/Aff_transformationH3.h

```diff
@@ -407,9 +411,9 @@ class Translation_repH3 : public Aff_transformation_rep_baseH3<R_>
   virtual  FT
            cartesian(int i, int j) const ;
 
-  Aff_transformation_3 compose(const Aff_transformation_rep_baseH3* aff) const
+  Aff_transformation_3 compose(const Base* aff) const
   {
-    const Translation_repH3* sr = dynamic_cast<const Translation_repH3*>(aff);
+    const Self* sr = dynamic_cast<const Self*>(aff);
```

Comment:
Shouldn't you check for null in-case the user supplies the wrong type?

`CGAL_precondition(sr != 0)`